crates/sui-source-validation/src/lib.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{BTreeMap, HashMap},
+    fmt::{Debug, Display},
+    str::FromStr,
+};
+
+use move_compiler::compiled_unit::CompiledUnitEnum;
+use move_core_types::account_address::AccountAddress;
+use move_package::compilation::compiled_package::CompiledPackage;
+use move_symbol_pool::Symbol;
+
+use sui_sdk::{
+    rpc_types::{SuiRawData, SuiRawMoveObject, SuiRawMovePackage},
+    ReadApi,
+};
+use sui_types::{
+    base_types::{ObjectID, ObjectIDParseError},
+    error::SuiError,
+};
+
+#[derive(Clone, Debug)]
+pub struct DependencyVerificationResult {
+    pub verified_dependencies: HashMap<AccountAddress, Dependency>,
+}
+
+#[derive(Debug)]
+pub enum DependencyVerificationError {
+    /// Could not convert a dependencies' resolved Sui address to a Sui object ID
+    ObjectIdFromAddressFailure(ObjectIDParseError),
+    /// Could not read a dependencies' on-chain object
+    DependencyObjectReadFailure(anyhow::Error),
+    /// Dependency object does not exist or was deleted
+    SuiObjectRefFailure(SuiError),
+    /// Dependency address contains a Sui object, not a Move package
+    ObjectFoundWhenPackageExpected(ObjectID, SuiRawMoveObject),
+    /// A local dependency was not found
+    ///
+    /// params:  package, module
+    LocalDependencyNotFound(Symbol, Option<Symbol>),
```

Comment:
Still have a preference here (and in examples below) that if the names of the params are worth documenting, it's better to document it in the code, rather than in a comment, but will leave it up to you:
```suggestion
    /// A local dependency was not found
    LocalDependencyNotFound { package: Symbol, module: Option<Symbol> },
```